datafusion/sqllogictest/test_files/array.slt

```diff
@@ -907,6 +927,16 @@ select list_push_back(make_array(1, 2, 3), 4), list_push_back(make_array(1.0, 2.
 ----
 [1, 2, 3, 4] [1.0, 2.0, 3.0, 4.0] [h, e, l, l, o]
 
+query ?
+select array_append([1, 2, 3], null);
```

Comment:
Can we also please add a test with more than two types, for example:

```sql
select [1, '2', Null]
```

Which postgres does

```sql
postgres=# select ARRAY [ 1, '2', NULL ];
   array
------------
 {1,2,NULL}
(1 row)
```

Reply:
It seems I would need casting between scalar value

ScalarValue I64(1) ->  ScalarValue F64(1.0)